core/src/main/scala/org/apache/spark/sql/hive/HiveClientUtil.scala

```diff
@@ -154,18 +154,6 @@ class HiveClientUtil(val sparkContext: SparkContext) extends Logging {
         dbDriver)
       metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME,
         Misc.SNAPPY_HIVE_METASTORE)
-    } else if (dbURL != null) {
-      logInfo(s"Using specified metastore database, dbURL = $dbURL")
-      metadataConf.setVar(HiveConf.ConfVars.METASTORECONNECTURLKEY, dbURL)
-      if (dbDriver != null) {
-        metadataConf.setVar(HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER,
-          dbDriver)
-      } else {
-        metadataConf.unset(
-          HiveConf.ConfVars.METASTORE_CONNECTION_DRIVER.varname)
-      }
-      metadataConf.unset(
-        HiveConf.ConfVars.METASTORE_CONNECTION_USER_NAME.varname)
     } else {
       logInfo("Using Hive metastore database, dbURL = " +
```

Comment:
Yes can remove those modes. As I had mentioned elsewhere the "embedded" property should be removed so then ExternalEmbeddedMode should be gone. The ExternalClusterMode is also unusable.